🐛 Fix uploads of .ogg that use application/ogg

Sweeistaken opened the pull request:
Firefox uploads .ogg files as application/ogg unlke Chromium-based browsers that upload as audio/ogg

A friend of mine uses Ghost and Firefox and has this problem but nobody reported an issue already, so I'd love to fix the issue with a PR alone.

This is one of my first contributions to a big project like this, feel free to suggest changes :3

- [x] I've read and followed the [Contributor Guide](https://github.com/TryGhost/Ghost/blob/main/.github/CONTRIBUTING.md)
- [x] I've explained my change
- [x] I've written an automated test to prove my change works

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> **Low Risk**
> Low risk config-only change that expands the allowed MIME types for media uploads, plus an e2e test to prevent regressions.
> 
> **Overview**
> Fixes Firefox `.ogg` upload failures by allowing `application/ogg` as a valid media upload content type.
> 
> Adds an e2e API test covering upload of an `.ogg` file explicitly sent with `application/ogg`, ensuring the `media/upload` endpoint accepts it and returns the expected URL/ref.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 833bd8c60653d2c1154550ea0ed43fcf1a2966be. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

Sweeistaken committed:
🐛 Fix uploads of .ogg that use application/ogg

Firefox uploads .ogg files as application/ogg unlke Chromium-based browsers that upload as audio/ogg

Sweeistaken commented on ghost/core/test/e2e-api/admin/media.test.js:
```diff
@@ -76,6 +76,21 @@ describe('Media API', function () {
+            const res = await request.post(localUtils.API.getApiQuery('media/upload'))
+                .set('Origin', config.get('url'))
+                .expect('Content-Type', /json/)
+                .field('ref', 'https://ghost.org/sample_640x360.ogv')
```
Right... I will apply that in another commit

Sweeistaken committed:
Correct URL in testing suite

markstos reviewed (commented):
Peer-review from a follow Ghost user (not a Ghost core team member)

According to this page about the file extensions, "application/ogg" is a valid extension for content in the "Ogg" container:

https://wiki.xiph.org/MIME_Types_and_File_Extensions#.ogg_-_audio/ogg

So, this looks like a positive addition for supporting Ogg uploads and Firefox compatibility.

e2e tests are a bit slow, so I question whether a new e2e test should be added every time a MIME type is added to the array. There are already some other tests to cover that the system works such that MIME types from this list are accepted. That will up to Ghost team to decide if they want the additional coverage or not.

Sweeistaken committed:
Merge branch 'main' into ogg

Sweeistaken committed:
Merge branch 'main' into ogg

Sweeistaken commented:
when are they going to stop adding so many bots to this repository...

9larsons committed:
Merge branch 'main' into ogg

9larsons committed:
Fixed application/ogg upload test to use node:assert

- the test used should-style assertions, but should is no longer
  installed in ghost/core and media.test.js was migrated to node:assert
- as written the test threw a TypeError instead of asserting, so it
  could not catch the regression it was added to cover

9larsons reviewed (approved):
Thanks for adding this! Sorry for the long delay, we were deep into 6.0 at the time.

I've pushed updates to the test files since we've retired `should` from the codebase.

9larsons committed:
Merge branch 'main' into ogg

9larsons committed:
Merge branch 'main' into ogg

9larsons committed:
Updated importer content-type test for application/ogg

- adding application/ogg to the media upload allowlist makes
  ImportManager.getContentTypes() return 24 entries, tripping the
  hard-coded count assertion (23) in this test
- this test only began running in CI once the vitest migration pulled
  test/unit/server/data/** into the unit-test run, so the brittle count
  was never previously exercised against an upload-config change
